lightning/src/ln/channel.rs

```diff
@@ -1495,7 +1495,150 @@ pub(super) struct ChannelContext<SP: Deref> where SP::Target: SignerProvider {
+			encode::serialize_hex(&funding_script), &self.context().channel_id());
+		secp_check!(self.context().secp_ctx.verify_ecdsa(&sighash, sig, self.context().counterparty_funding_pubkey()), format!("Invalid {} signature from peer", self.received_msg()));
+
+		Ok(initial_commitment_tx)
+	}
+
+	fn initial_commitment_signed<L: Deref>(
+		&mut self, channel_id: ChannelId, counterparty_signature: Signature, counterparty_txid: Txid,
+		counterparty_commitment_number: u64, best_block: BestBlock, signer_provider: &SP, logger: &L,
+	) -> Result<ChannelMonitor<<SP::Target as SignerProvider>::EcdsaSigner>, ChannelError>
+	where
+		L::Target: Logger
+	{
+		let initial_commitment_tx = match self.check_counterparty_commitment_signature(&counterparty_signature, logger) {
+			Ok(res) => res,
+			Err(ChannelError::Close(e)) => {
+				self.context_mut().channel_transaction_parameters.funding_outpoint = None;
+				return Err(ChannelError::Close(e));
+			},
+			Err(e) => {
+				// The only error we know how to handle is ChannelError::Close, so we fall over here
+				// to make sure we don't continue with an inconsistent state.
+				panic!("unexpected error type from check_counterparty_commitment_signature {:?}", e);
+			}
+		};
+		let counterparty_keys = self.context().build_remote_transaction_keys();
+		let counterparty_initial_commitment_tx = self.context().build_commitment_transaction(self.context().cur_counterparty_commitment_transaction_number, &counterparty_keys, false, false, logger).tx;
+		let counterparty_trusted_tx = counterparty_initial_commitment_tx.trust();
+		let counterparty_initial_bitcoin_tx = counterparty_trusted_tx.built_transaction();
+
+		log_trace!(logger, "Initial counterparty tx for channel {} is: txid {} tx {}",
+			&self.context().channel_id(), counterparty_initial_bitcoin_tx.txid, encode::serialize_hex(&counterparty_initial_bitcoin_tx.transaction));
+
+		let holder_commitment_tx = HolderCommitmentTransaction::new(
+			initial_commitment_tx,
+			counterparty_signature,
+			Vec::new(),
+			&self.context().get_holder_pubkeys().funding_pubkey,
+			self.context().counterparty_funding_pubkey()
+		);
+
+		if self.context().holder_signer.as_ref().validate_holder_commitment(&holder_commitment_tx, Vec::new()).is_err() {
+			return Err(ChannelError::close("Failed to validate our commitment".to_owned()));
+		}
+
+		// Now that we're past error-generating stuff, update our local state:
+
+		self.context_mut().channel_id = channel_id;
+
+		assert!(!self.context().channel_state.is_monitor_update_in_progress()); // We have not had any monitor(s) yet to fail update!
+		if self.context().is_batch_funding() {
+			self.context_mut().channel_state = ChannelState::AwaitingChannelReady(AwaitingChannelReadyFlags::WAITING_FOR_BATCH);
+		} else {
+			self.context_mut().channel_state = ChannelState::AwaitingChannelReady(AwaitingChannelReadyFlags::new());
+		}
+		{
+			let context = self.context_mut();
```

Comment:
If we move this after `initial_commitment_tx` is initialized, we can use that variable throughout instead of calling `self.context()` and `self.context_mut()` everywhere. Could drop the scope used here, too.